### ISSUE TITLE
'expo-auth-session/providers/google' broken in prod/play store?

### ISSUE THREAD

**Issue by ariccio** [expo/expo#16650]
### Summary

It seems like something is broken, and I'm absolutely mystified what or why. Google may have changed something?

I'm using a bare workflow app on android. Google sign in works fine in development. In the store build, as downloaded, I get this error: 
![Screenshot_20220315-170027](https://user-images.githubusercontent.com/2142308/158475869-0d7de07b-2387-4a4f-af78-41092e9fd0e4.png)

A google search for the exact phrase `"You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure."` turns up very few results, but it looks like a bunch are very new - the last few weeks.

[You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure](https://stackoverflow.com/q/70638732/625687)
[Is there a way to override the CallbackPath to include the scheme?](https://stackoverflow.com/q/69933339/625687)
[Nest Authorization Error Error 400: invalid_request](https://community.home-assistant.io/t/nest-authorization-error-error-400-invalid-request/399388)
[Support » Plugin: FluentSMTP - WP Mail SMTP, Amazon SES, SendGrid, MailGun and Any SMTP Connector Plugin » Error 400: invalid_request](https://wordpress.org/support/topic/error-400-invalid_request-2/)
[Adding Nest integration fails: Error 400: invalid_request](https://www.reddit.com/r/homeassistant/comments/t9xnrk/adding_nest_integration_fails_error_400_invalid/)

Incidentally, it appears a few of those have to do with Google's recent oauth changes ([Making Google OAuth interactions safer by using more secure OAuth flows](https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#dates-oob))

I don't seem to have *that* problem. My package name in the cloud console (`com.ariccio.co2_native_client`) matches the manifest (`"com.ariccio.co2_native_client"`), which I just copied and pasted to confirm. The displayed redirect URI seems to match [the requirements for `redirect_uri`](https://developers.google.com/identity/protocols/oauth2/native-app#step-2:-send-a-request-to-googles-oauth-2.0-server), and it's not returning `redirect_uri_mismatch` anyways. I also have the following schemes, I don't think that's the problem:
```
        <data android:scheme="com.ariccio.co2_native_client"/>
        <data android:scheme="com.ariccio.co2_native_client:/oauthredirect"/>

```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### Environment

  expo-env-info 1.0.2 environment info:
    System:
      OS: Windows 10 10.0.19044
    Binaries:
      Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.15 - C:\Program Files\nodejs\yarn.CMD
      npm: 8.5.4 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version     2021.1.0.0 AI-211.7628.21.2111.8193401
    npmPackages:
      expo: ~44.0.0 => 44.0.6
      react: 17.0.2 => 17.0.2
      react-dom: 17.0.2 => 17.0.2
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.7 => 0.17.7
    Expo Workflow: bare

### Reproducible demo

Incidentally, it looks like if I just copy and paste the google example text for expo go into a snack and create a clientid, well, it doesn't work?!?

https://snack.expo.dev/@ariccio/expo-auth-session-test

Is this related to the open bug: https://github.com/expo/expo/issues/12044

**Comment by ariccio** [comment]
It appears to have something to do with the combination of my package name and the package name in the google cloud console. Investigating further.

**Comment by ariccio** [comment]
Ok, I believe I have ruled out an SDK problem. Closing.

**Comment by azolyomi** [comment]
@ariccio did you ever figure out the root cause of this? I am struggling as well.

**Comment by ariccio** [comment]
No. This was, by far, the most INFURIATING bug of the year. I suspect there was some kind of backend problem somewhere, since I'd get an occasional opaque error (lol yay Google) in the cloud console.

The only way I worked around it was to switch the app/package name back to `riccio.co2.client`. It feels weird but honestly the whole java convention of naming things and putting them in deeply nested folders is pretty stupid imo, so I have survived 🤣
